tests/pfc/test_pfc_pause_lossy.py

```diff
@@ -0,0 +1,106 @@
+                                 port_bandwidth,
+                                 frame_size) :
+    """
+    This test case checks the behaviour of the SONiC DUT when it receives 
+    a PFC pause frame on lossy priorities.
+
+                                +-----------+
+    [Keysight Chassis Tx Port]  |           | [Keysight Chassis Rx Port]
+    --------------------------->| SONiC DUT |<---------------------------
+    Test Data Traffic +         |           |  PFC pause frame on 
+    Background Dada Traffic     +-----------+  "lossy" priorities.
+
+    1. Configure SONiC DUT with multipul lossless priorities. 
+    2. On SONiC DUT enable PFC on several lossless priorities e.g priority 
+       3 and 4.
+    3. On the Keysight chassis Tx port create two flows - a) 'Test Data Traffic'
+       and b) 'Background Data traffic'.
+    4. Configure 'Test Data Traffic' such that it contains traffic items
+       with all lossy priorities.
+    5. Configure 'Background Data Traffic' it contains traffic items with
+       all lossless priorities.
+    6. From Rx port send pause frames on all lossless priorities. Then
+       start 'Test Data Traffic' and 'Background Data Traffic'.
+    7. Verify the following: 
+       (a) When Pause Storm are running, Keysight Rx port is receiving
+       both 'Test Data Traffic' and 'Background Data traffic'.
+       (b) When Pause Storm are stoped, then also Keysight Rx port is receiving
+       both 'Test Data Traffic' and 'Background Data traffic'.
+    """
+    duthost.shell('sudo pfcwd stop')
+
+    for base_config in lossy_configs:
+        # create the configuration
+        api.set_state(State(ConfigState(config=base_config, state='set')))
+
+        # start all flows
+        api.set_state(State(FlowTransmitState(state='start')))
+
+        exp_dur = start_delay + traffic_duration
+        logger.info("Traffic is running for %s seconds" %(exp_dur))
+        time.sleep(exp_dur)
+
+        # stop all flows
+        api.set_state(State(FlowTransmitState(state='stop')))
+
+        # Get statistics
+        stat_captions =['Test Data', 'Background Data']
+        for row in api.get_flow_results(FlowRequest(flow_names=stat_captions)):
+            if (row['name'] == 'Test Data') or (row['name'] == 'Background Data'):
+
+                if ((row['frames_rx'] == 0) or (row['frames_tx'] != row['frames_rx'])):
+                     pytest_assert(False, 
+                         "Not all %s reached Rx End" %(rows[caption_index]))
+
+                line_rate = traffic_line_rate / 100.0
+                exp_rx_bytes = (port_bandwidth * line_rate * traffic_duration) / 8
+                tolerance_ratio = row['bytes_rx'] / exp_rx_bytes
+    
+                if ((tolerance_ratio < TOLERANCE_THRESHOLD) or
+                    (tolerance_ratio > 1)) :
+                    pytest_assert(False,
+                        "expected % of packets not received at the RX port")
```

Comment:
Please don't use pytest_assert(False, msg) (see comments above)

Reply:
I have replaced this pytest_assert(False, ...) with pytest.fail(msg)